test/integration/master/audit_dynamic_test.go

```diff
@@ -0,0 +1,263 @@
+		configMapOperations(t, kubeclient)
+
+		// check for corresponding events
+		missing, err := checkForEvents(t, eventList1)
+		require.NoError(t, err, "Failed to match all expected events for sink1, events %#v not found!", missing)
+		require.Len(t, eventList2.Load().(auditinternal.EventList).Items, 0, "eventList2 should be empty")
+	})
+
+	t.Run("update sink", func(t *testing.T) {
+		sink1, err := kubeclient.AuditregistrationV1alpha1().AuditSinks().Get("test1", metav1.GetOptions{})
+		require.NoError(t, err)
+		sink1.Spec.Webhook.ClientConfig.URL = &server2.URL
+		_, err = kubeclient.AuditregistrationV1alpha1().AuditSinks().Update(sink1)
+		require.NoError(t, err, "failed to update audit sink1")
+		t.Log("updated audit sink1")
+
+		// sleep to wait for sink to close
+		time.Sleep(5 * time.Second)
+		eventList1.Store(auditinternal.EventList{})
+		eventList2.Store(auditinternal.EventList{})
+
+		// perform configmap ops
+		configMapOperations(t, kubeclient)
+
+		// check for corresponding events
+		missing, err := checkForEvents(t, eventList2)
+		require.NoError(t, err, "Failed to match all expected events for sink2, events %#v not found!", missing)
+		require.Len(t, eventList1.Load().(auditinternal.EventList).Items, 0, "eventList1 should be empty")
+	})
+}
+
+// checkForEvents will poll to check for the expected audit events in an atomic event list
+func checkForEvents(t *testing.T, a *atomic.Value) ([]utils.AuditEvent, error) {
+	var missing []utils.AuditEvent
+	err := wait.Poll(1*time.Second, 120*time.Second, func() (bool, error) {
+		var err error
+		el := a.Load().(auditinternal.EventList)
+		if len(el.Items) < 1 {
+			return false, nil
+		}
+		missing, err = utils.CheckAuditList(el, expectedEvents)
+		require.NoError(t, err, "problem checking audit list")
+		if len(missing) > 0 {
+			return false, nil
+		}
+		t.Logf("all events found")
+		return true, nil
+	})
+	return missing, err
+}
+
+// buildTestHandler returns a handler that will update the atomic value passed in
+// with the event list it receives
+func buildTestHandler(t *testing.T, a *atomic.Value) http.HandlerFunc {
+	decoder := audit.Codecs.UniversalDecoder(auditv1.SchemeGroupVersion)
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		body, err := ioutil.ReadAll(r.Body)
+		require.NoError(t, err, "could not read request body")
+		el := auditinternal.EventList{}
+		err = runtime.DecodeInto(decoder, body, &el)
+		require.NoError(t, err, "failed decoding buf: %b, apiVersion: %s", body, auditv1.SchemeGroupVersion)
+		defer r.Body.Close()
+		cl := a.Load().(auditinternal.EventList)
+		el.Items = append(el.Items, cl.Items...)
+		a.Store(el)
+		w.WriteHeader(200)
+	})
+}
+
+// testServerHealthCheck tests if the given server url is capable of storing audit events
+func testServerHealthCheck(url string, encoder runtime.Encoder) error {
+	err := wait.Poll(100*time.Millisecond, 10*time.Second, func() (bool, error) {
+		el := auditinternal.EventList{}
```

Comment:
You could use a mux for your test server and add a `/healthz` route that returns 200 for a `GET` request, instead of doing this